ports/espressif/Makefile

```diff
@@ -163,12 +163,9 @@ ifeq ($(DEBUG), 1)
   # CFLAGS += -fno-inline -fno-ipa-sra
 else
   CFLAGS += -DNDEBUG -ggdb3
-	ifeq ($(IDF_TARGET_ARCH),xtensa)
-	  OPTIMIZATION_FLAGS ?= -O2
-	else
-		# RISC-V is larger than xtensa so do -Os for it
-	  OPTIMIZATION_FLAGS ?= -Os
-	endif
+  OPTIMIZATION_FLAGS ?= -O2
+  # RISC-V is larger than xtensa
+  # Use -Os for RISC-V when it overflows
```

Comment:
Did you find this necessary to prevent boot-loops? If we went to -Os would it stop working?

Reply:
The optimization level doesn't affect boot-loops. We have `~30K` headroom, so I switched to `O2` for now.